src/lib.rs

```diff
@@ -93,19 +93,28 @@ impl CommandTrigger {
 
         let (instance, mut store) = instance_builder.instantiate(()).await?;
 
-        let func = {
-            let instance_export = instance
-                .get_export(&mut store, None, "wasi:cli/run@0.2.0")
-                .context("failed to find the wasi:cli/run@0.2.0 instance in component")?;
-
-            let func_export = instance
-                .get_export(&mut store, Some(&instance_export), "run")
-                .context("failed to find the \"run\" function in wasi:cli/run@0.2.0 instance")?;
-
-            instance
-                .get_typed_func::<(), (Result<(), ()>,)>(&mut store, func_export)
-                .context("failed to get typed \"run\" function")?
-        };
+        let versions = ["0.2.3", "0.2.2", "0.2.1", "0.2.0"];
```

Comment:
@alexcrichton @dicej Is this really what hosts have to do to support multiple supposedly compatible 0.2.x versions?  It's not a big deal here but suboptimal in more performance-sensitive scenarios (and a bit excessive in terms of ceremony around the core notion of "we call the P2 `run` function").

Reply:
I actually had a completely unrelated refactor I just did at https://github.com/spinframework/spin/pull/3063 which would solve this problem here too.

In short, no, versions should not need to be explicitly probed for (they're going to infinitely increase and we don't want an infinitely growing list here).

Here what I would recommend is to use [`Command::new`](https://docs.rs/wasmtime-wasi/latest/wasmtime_wasi/bindings/struct.Command.html#method.new) passing in `&mut store` and `&instance`. If that's successful you can call `the_ret.wasi_cli_run().run(&mut store)`. That'll internally handle versions, typechecks, etc.